lib/postgrex/pgpass.ex

```diff
@@ -47,17 +36,45 @@ defmodule Postgrex.Pgpass do
     |> Enum.map(&(String.split(&1, ":")))
   end
 
-  defp readable_pgpass_file? do
-    with {:ok, stat} <- File.stat(pgpass_path()),
+  defp readable_pgpass_file?(nil) do
+    # when the pgpass is not passed via the :passfile option, simply ignore and move on
+    with default_path <- pgpass_path(),
+         {:ok, stat} <- File.stat(default_path),
          0o0600 <- stat.mode &&& 0o0777,
-    do: true,
+    do: {:ok, default_path},
     else: (_ -> false)
   end
+  defp readable_pgpass_file?(passfile) do
+    # when using the :passfile options, the file must be readable or an error is raised
+    case File.stat(passfile) do
```

Comment:
We could use `File.stat!` so we don't need to raise all these `RuntimeError`s. Note that `:file.format_error` can format errors but `FileError` would be clearer I think.